### PR TITLE
samples(grid): Adding extenral filtering samples #1546

### LINKED ISSUE #1546: External Excel Style and Advanced filtering dialogs samples
Related to IgniteUI/igniteui-angular#5981

### PR DESCRIPTION
**Pull request by astaev**
Closes #1546

### CODE REVIEW

File: src/app/hierarchical-grid/hierarchical-grid-external-excel-style-filtering/hierarchical-grid-external-excel-style-filtering.component.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<div class="wrapper">
+    <igx-hierarchical-grid class="hgrid" [data]="localdata" [showToolbar]="true" [columnHiding]="true" [autoGenerate]="false" [allowFiltering]='true' filterMode="excelStyleFilter"
```
Comment (DiyanDimitrov) [inline src/app/hierarchical-grid/hierarchical-grid-external-excel-style-filtering/hierarchical-grid-external-excel-style-filtering.component.html]: Disable built-in filtering.

File: src/app/hierarchical-grid/hierarchical-grid-external-advanced-filtering/hierarchical-grid-external-advanced-filtering.component.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<div class="grid__wrapper">
+    <igx-hierarchical-grid class="hgrid" [data]="localdata" [autoGenerate]="false" [showToolbar]="true"
+        [allowAdvancedFiltering]="true" [height]="'500px'" [width]="'800px'" [rowHeight]="'65px'" #hierarchicalGrid>
```
Comment (DiyanDimitrov) [inline src/app/hierarchical-grid/hierarchical-grid-external-advanced-filtering/hierarchical-grid-external-advanced-filtering.component.html]: Disable built-in advanced filtering.

File: src/app/tree-grid/tree-grid-external-advanced-filtering/tree-grid-external-advanced-filtering.component.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<div class="flex-row">
+    <igx-tree-grid #treegrid1 [data]="data" [autoGenerate]="false" height="500px" width="100%" [allowFiltering]="true" 
+        primaryKey="ID" foreignKey="ParentID" filterMode="excelStyleFilter">
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-external-advanced-filtering/tree-grid-external-advanced-filtering.component.html]: Disable built-in Excel style filtering.

File: src/app/grid/grid-external-advanced-filtering/grid-external-advanced-filtering.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        this.data = DATA;
+    }
+
+    public formatDate(val) {
+        if (val !== "Select All") {
+            return new Intl.DateTimeFormat("en-US").format(val);
+        } else {
+            return val;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (astaev) [reply src/app/grid/grid-external-advanced-filtering/grid-external-advanced-filtering.component.ts]: I guess no, was copied from next topic.

File: src/app/tree-grid/tree-grid-routes-data.ts [hunk 5]
```diff
@@ -45,6 +45,10 @@ export const treeGridRoutesData = {
     },
     "treegrid-excel-style-filtering-load-on-demand":
         { displayName: "TreeGrid Excel Style Filtering Load On Demand", parentName: "TreeGrid" },
+    "tree-grid-external-excel-style-filtering":
+        { displayName: "TreeGrid Extenral Excel Style Filtering", parentName: "TreeGrid" },
+    "tree-grid-external-advanced-filtering":
+        { displayName: "TreeGrid Extenral Advanced Filtering", parentName: "TreeGrid" },
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-routes-data.ts]: There is a typo in **Extenral**.

File: src/app/grid/grid-external-advanced-filtering/grid-external-advanced-filtering.component.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<div class="grid__wrapper">
+    <div class="advanced-dialog">
+        <igx-advanced-filtering-dialog [grid]="grid1" >
+        </igx-advanced-filtering-dialog>
+    </div>
```
[the remark this answers is not in the document]
Comment (astaev) [reply src/app/grid/grid-external-advanced-filtering/grid-external-advanced-filtering.component.html]: Done

File: src/app/tree-grid/tree-grid-routes-data.ts (hunk 5 above)
Comment (astaev) [reply src/app/tree-grid/tree-grid-routes-data.ts]: Done

File: src/app/tree-grid/tree-grid-external-advanced-filtering/tree-grid-external-advanced-filtering.component.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
+<div class="grid__wrapper">
+    <igx-advanced-filtering-dialog [grid]="treegrid1" class="advanced-dialog">
+    </igx-advanced-filtering-dialog>
+
+    <igx-tree-grid #treegrid1 [data]="data" [autoGenerate]="false" height="400px" width="100%" [allowFiltering]="true"
```
Comment (DiyanDimitrov) [inline src/app/tree-grid/tree-grid-external-advanced-filtering/tree-grid-external-advanced-filtering.component.html]: Set grid's width to 800px.

**Comment by DiyanDimitrov** [comment]
There is a build error:
src/app/hierarchical-grid/hierarchical-grid-external-excel-style-filtering/hierarchical-grid-external-excel-style-filtering.component.html(24,71): Property 'formatter' does not exist on type 'HGridExternalExcelStyleFilteringComponent'.

File: src/app/tree-grid/tree-grid-external-advanced-filtering/tree-grid-external-advanced-filtering.component.html (hunk 7 above)
Comment (astaev) [reply src/app/tree-grid/tree-grid-external-advanced-filtering/tree-grid-external-advanced-filtering.component.html]: Done

**Comment by astaev** [comment]
Build error fixed

**Comment by DiyanDimitrov** [comment]
TreeGrid External Advanced Filtering sample is not working is Stackblitz:
_Error in /~/src/app/app.module.ts (23:17)
TreeGridExternalAdvancedFilteringComponent is not defined_

**Pull request merged**